src/main/java/com/schibsted/spt/data/jslt/impl/PipeOperator.java

```diff
@@ -0,0 +1,39 @@
+
+// Copyright 2018 Schibsted Marketplaces Products & Technology As
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.schibsted.spt.data.jslt.impl;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import com.schibsted.spt.data.jslt.JsltException;
+
+public class PipeOperator extends AbstractOperator {
+
+  public PipeOperator(ExpressionNode left, ExpressionNode right,
+                      Location location) {
+    super(left, right, "|", location);
+  }
+
+  @Override
+  public JsonNode apply(Scope scope, JsonNode input) {
+    right.computeMatchContexts(new DotExpression(new Location(null, 0,0)));
```

Comment:
We want to do this at compile time instead of repeating it every time the expression is evaluated. It's better to implement it in the `computeMatchContexts` method. Note that you need to do it for both `left` and `right`.

Reply:
Basically, the inherited logic in `computeMatchContexts` is correct for `left`, but you need this line to handle `right`

Reply:
Right, I addressed this in the last commit. Take a look at the latest one.